src/pyrenew/process/differencedprocess.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+
+from __future__ import annotations
+
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+from numpyro.contrib.control_flow import scan
+
+from pyrenew.metaclass import RandomVariable, SampledValue
+
+
+class DifferencedProcess(RandomVariable):
+    """
+    Class for differenced stochastic process X(t),
+    constructed by placing a fundamental stochastic
+    process on the :math:`n^{th}` differences
+    (rates of change). See
+    https://otexts.com/fpp2/stationarity.html
+    for a discussion of differencing in the
+    context of discrete timeseries data.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        fundamental_process: RandomVariable,
+        differencing_order: int,
+        **kwargs,
+    ) -> None:
+        """
+        Default constructor
+
+        Parameters
+        ----------
+        name : str
+            Name of the stochastic process
+        fundamental_process : RandomVariable
+            Stochastic process for the
+            first differences. Should accept an
+            `n` argument specifying the number
+            of samples to draw.
+        differencing_order : int
+            How many fold-differencing the
+            the process represents. Must be
+            an integer greater than or
+            equal to 1. 1 represents a process
+            on the first differences (the rate
+            of change), 2 a process on the
+            2nd differences (rate of change of
+            the rate of change), et cetera.
+        **kwargs :
+            Additional keyword arguments passed to
+            the parent class constructor.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        The order of differencing is the discrete
+        analogue of the order of a derivative in single
+        variable calculus. A first difference (derivative)
+        represents a rate of change. A second difference
+        (derivative) represents the rate of change of that
+        rate of change, et cetera.
+        """
+        self.assert_valid_differencing_order(differencing_order)
```

Comment:
I think we have many checks for "integer greater than 1" throughout the codebase. Maybe a common function would be helpful.

Reply:
Agree. Separate issue I think.